tests/models/compare_utils.py

```diff
@@ -1,33 +1,68 @@
-"""Compare the logprobs of two sequences generated by different models, 
+"""Compare the logprobs of two sequences generated by different models,
 which should be similar but not necessarily equal.
 """
 
+import pandas as pd
+
 
 def check_logprobs_close(outputs_0_lst, outputs_1_lst, name_0, name_1):
+
     # Loop through responses to each prompt.
+    prompts_in_error = dict()
     for prompt_idx, (outputs_0,
                      outputs_1) in enumerate(zip(outputs_0_lst,
                                                  outputs_1_lst)):
-        output_ids_0, output_str_0, logprobs_0 = outputs_0
-        output_ids_1, output_str_1, logprobs_1 = outputs_1
-
-        # Loop through generated tokens.
-        for idx, (output_token_0,
-                  output_token_1) in enumerate(zip(output_ids_0,
-                                                   output_ids_1)):
-
-            # If generated tokens don't match, then
-            if output_token_0 != output_token_1:
-                # Each predicted token must be in top N logprobs of the other
-                assert output_token_0 in [
-                    s.strip() for s in logprobs_1[idx]
-                ], (f"Test{prompt_idx}:"
-                    f"\n{name_0}:\t{output_token_0!r}"
-                    f"\n{name_1}:\t{output_token_1!r}")
-                assert output_token_1.strip() in logprobs_0[idx], (
-                    f"Test{prompt_idx}:"
-                    f"\n{name_0}:\t{output_token_0!r}"
-                    f"\n{name_1}:\t{output_token_1!r}")
-
-                # Break out since sequences will now diverge.
-                break
+        # combine the data into a single dataframe
+        outputs_0_df = pd.DataFrame.from_records(outputs_0).transpose().drop(
+            columns=1).dropna(axis=0,
+                              how="all").rename(columns={
+                                  0: f"{name_0}_tokens_index",
+                                  2: f"{name_0}_keyed_logprobs"
+                              })
+        outputs_1_df = pd.DataFrame.from_records(outputs_1).transpose().drop(
+            columns=1).dropna(axis=0,
+                              how="all").rename(columns={
+                                  0: f"{name_1}_tokens_index",
+                                  2: f"{name_1}_keyed_logprobs"
+                              })
+        all_outputs_df = pd.concat([outputs_0_df, outputs_1_df],
+                                   axis="columns")
+
+        # find name_0 tokens that appear in any of the keys of the name_1
+        # logprobs
+        all_outputs_df[
+            f"{name_0}_token_in_{name_1}_logprobs"] = all_outputs_df.apply(
+                lambda row: any(kee.strip() in row[f"{name_0}_tokens_index"]
```

Comment:
I did not intend to keep these calls to `.strip()`.  it was an experiment to see what the results would be like w/ the leading spaces missing.